landing_page_app/main/routes/auth.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import os
+from urllib.parse import quote_plus, urlencode
+
+from authlib.integrations.flask_client import OAuth
+from flask import Blueprint, current_app, redirect, render_template, session, url_for
+
+from landing_page_app.main.config.constants import (
+    AS_ORG_ALLOWED_EMAIL_DOMAINS,
+    MINISTRY_OF_JUSTICE,
+    MOJ_ANALYTICAL_SERVICES,
+    MOJ_ORG_ALLOWED_EMAIL_DOMAINS,
+    MOJ_ORGS,
+)
+
+logger = logging.getLogger(__name__)
+
+AUTHLIB_CLIENT = "authlib.integrations.flask_client"
+
+auth_route = Blueprint("auth_routes", __name__)
+
+
+@auth_route.record
+def setup_auth0(setup_state):
+    app = setup_state.app
+    OAuth(app)
+    auth0 = app.extensions.get(AUTHLIB_CLIENT)
+    auth0.register(
+        "auth0",
+        client_id=os.getenv("AUTH0_CLIENT_ID"),
+        client_secret=os.getenv("AUTH0_CLIENT_SECRET"),
+        client_kwargs={
+            "scope": "openid profile email",
+        },
+        server_metadata_url=f'https://{os.getenv("AUTH0_DOMAIN")}/'
+        + ".well-known/openid-configuration",
+    )
+
+
+@auth_route.route("/login")
+def login():
+    logger.debug("login()")
+    auth0 = current_app.extensions.get(AUTHLIB_CLIENT)
+    return auth0.auth0.authorize_redirect(
+        redirect_uri=url_for("auth_routes.callback", _external=True, _scheme="https")
```

Comment:
Adding `_scheme="https"` is the non-ideal fix for enforcing https on the callback